tools/fast/fasting/basic.go

```diff
@@ -0,0 +1,132 @@
+package fasting
+
+import (
+	"context"
+	"io/ioutil"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-ipfs-files"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/testhelpers"
+	"github.com/filecoin-project/go-filecoin/tools/fast"
+	"github.com/filecoin-project/go-filecoin/tools/fast/series"
+	localplugin "github.com/filecoin-project/go-filecoin/tools/iptb-plugins/filecoin/local"
+)
+
+type Environment struct {
+	fast.Environment
+
+	t   *testing.T
+	ctx context.Context
+
+	pluginName string
+	pluginOpts map[string]string
+
+	fastenvOpts fast.EnvironmentOpts
+
+	GenesisMiner *fast.Filecoin
+}
+
+func (env *Environment) NewNode() *fast.Filecoin {
+	require := require.New(env.t)
+
+	p, err := env.NewProcess(env.ctx, env.pluginName, env.pluginOpts, env.fastenvOpts)
+	require.NoError(err)
+
+	return p
+}
+
+func (env *Environment) NewNodeStarted() *fast.Filecoin {
+	require := require.New(env.t)
+
+	p := env.NewNode()
+
+	err := series.InitAndStart(env.ctx, p)
+	require.NoError(err)
+
+	return p
+}
+
+func (env *Environment) NewNodeConnected() *fast.Filecoin {
+	require := require.New(env.t)
+
+	p := env.NewNodeStarted()
+
+	err := series.Connect(env.ctx, env.GenesisMiner, p)
+	require.NoError(err)
+
+	return p
+}
+
+func (env *Environment) NewNodeWithFunds(funds int) *fast.Filecoin {
+	require := require.New(env.t)
+
+	p := env.NewNodeConnected()
+
+	err := series.SendFilecoinDefaults(env.ctx, env.GenesisMiner, p, funds)
+	require.NoError(err)
+
+	return p
+}
+
+// BasicFastSetup creates a environment and a single node, and environment options
+func BasicFastSetup(ctx context.Context, t *testing.T, fastenvOpts fast.EnvironmentOpts) (*Environment, context.Context) {
+	require := require.New(t)
+
+	// Create a directory for the test using the test name (mostly for FAST)
+	// Replace the forward slash as tempdir can't handle them
+	dir, err := ioutil.TempDir("", strings.Replace(t.Name(), "/", ".", -1))
```

Comment:
I think devs would find FAST tests easier to debug if we put this repo in the same directory as the test being ran. I believe this is what go-ipfs does for sharness tests.

Reply:
Any idea how to get that?

Reply:
oops, I didn't realize how complicated this ask was. Yes, but it's hacky and involves the runtime package -- probably better not to do it.

...But [this](https://github.com/sirupsen/logrus/blob/a6c0064cfaf982707445a1c90368f956421ebcf0/entry.go#L166-L196) is what I would reference if I were to attempt it